src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -185,17 +151,21 @@ public override bool Execute ()
         if (!InvariantGlobalization)
             nativeAssets.Add(IcuDataFileName!);
 
-        foreach (var f in nativeAssets)
-            File.Copy(Path.Join (runtimeSourceDir, f), Path.Join(AppDir, f), true);
-        File.Copy(MainJS!, Path.Join(AppDir, "runtime.js"),  true);
+        if (Path.TrimEndingDirectorySeparator(Path.GetFullPath(runtimeSourceDir)) != Path.TrimEndingDirectorySeparator(Path.GetFullPath(AppDir!)))
+        {
+            foreach (var f in nativeAssets)
+                FileCopyChecked(Path.Join(runtimeSourceDir, f), Path.Join(AppDir, f), "NativeAssets");
+        }
+        FileCopyChecked(MainJS!, Path.Join(AppDir, "runtime.js"), string.Empty);
 
         var html = @"<html><body><script type=""text/javascript"" src=""runtime.js""></script></body></html>";
```

Comment:
Should creating index.html be optional?  Would it make sense to be able to supply one?

Reply:
I'm not sure about it being optional. In what cases is it used, and not used?

But it would be helpful to be able to supply one. I have an upcoming PR which would be adding conditionally adding bits to the html.

Reply:
I think for both samples in `dotnet/runtime` and `dotnet/samples`, the html gets overwritten/supplied in the csprojs.